tendrl/integrations/ceph/sds_sync/__init__.py

```diff
@@ -0,0 +1,108 @@
+import gevent
+import json
+import os
+
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message, ExceptionMessage
+from tendrl.commons.utils import cmd_utils
+from tendrl.commons import sds_sync
+from tendrl.node_agent.node_sync import disk_sync
+
+
+class CephIntegrtaionsSyncThread(sds_sync.StateSyncThread):
+    def _run(self):
+        Event(
+            Message(
+                priority="info",
+                publisher=NS.publisher_id,
+                payload={"message": "%s running" % self.__class__.__name__}
+            )
+        )
+        while not self._complete.is_set():
+            # sync of OSD journal details every 30 sec is fine
+            # as this data doesnt change very frequently in cluster
+            gevent.sleep(30)
+            if NS.tendrl_context.integration_id is not None:
+                try:
+                    # Sync the OSD journal details
+                    Event(
+                        Message(
+                            priority="info",
+                            publisher=NS.publisher_id,
+                            payload={
+                                "message":
+                                "ceph_integrations_sync, osd journal details"
+                            }
+                        )
+                    )
+                    journal_details = {}
+                    if os.path.isdir(
+                        "/var/lib/ceph/osd/%s" % NS.tendrl_context.cluster_name
```

Comment:
Else part is taking of that only. It would list all the files under `/var/lib/ceph/osd` and the file names would be like `/var/lib/ceph/osd/{ClusterName}-{osd_id}`. Check lines 49,50 for the same.